### DISCUSSION TITLE
https://next-auth.js.org/warnings#nextauth_url NEXTAUTH_URL environment variable not set

### DISCUSSION THREAD

**Discussion by tomthor** [WebDevStudios/nextjs-wordpress-starter#795] [Q&A]
After 'npm run dev' I get the following warning: 

[next-auth][warn][nextauth_url] 
https://next-auth.js.org/warnings#nextauth_url NEXTAUTH_URL environment variable not set

This is strange as NEXTAUTH_URL is set in .env  - I have triple checked all the .env values from docs and linked the ones from wp-config. The project doesn't seem to work properly, i'm also receiving errors such as 

"To display your WordPress homepage dynamically, set your homepage to a static page via the WP dashboard (Settings: Reading Settings)." when i load the home page.

If I try to load a page I get - 

"Server Error
Error: Only absolute URLs are supported" 

I am using Laravel Valet, PHP@7.4 as a local server to host my Wordpress install. 

Any insight on these errors/warnings would be gladly received. I have spent so long trying to figure out what is causing this

Thanks

**Comment by itsamoreh** [comment]
@tomthor Hello! Thanks for taking a look at the starter!

Are you still experiencing this issue?

We've made some updates to the starter. Can you pull the latest version of `canary` and see if the frontend is still broken?

**Reply by arm-ns** [reply]
hey, @tomthor I had the same issue.  Try to change these strings (both in wp-config and .env) to something simple. It seems that the suggested method generate a random string with WordPress Salt generator sometimes generates the string with forbidden symbols that's why it breaks.

```
define( 'PREVIEW_SECRET_TOKEN', 'ANY_RANDOM_STRING');
define( 'GRAPHQL_JWT_AUTH_SECRET_KEY', 'ANY_RANDOM_STRING' );
```
